Topics/Development_Process/Github_Pages.md

```diff
@@ -0,0 +1,91 @@
+
+## Getting started - Deploying a static HTML site
+
+Here's a quick overview on **how to deploy a simple, static HTML site.** It's simple, but it can be hard to remember clearly, so feel free to use this as a reference every time you do so.
+
+1. Create a new repo.
+2. Add your HTML/CSS/js files to your repository.
+   1. You should name your HTML file `index.html` -- that's the "home" page.
+   2. Other than that, if your css/javascript works on your machine if you were to clone this repo, it should also work on GitHub pages.
+3. Go to your repository settings.
+4. Navigate to Pages in the repository's settings.
+5. Choose a branch you want to deploy your page to (it should say `None` by default`). **The moment you select your branch and hit save, GitHub Pages will be enabled and will start deploying.** It should take between 1-2 minutes. This point onwards, your GitHub page will update every time you commit something to the repository.
+
+Your site by default will have this URL:
+
+```
+<your_GitHub_username>.github.io/<repository_name>/
+```
+
+By default, `index.html` will show up.
+
+To access another file in your repository, your URL would be:
+
+```
+<your_GitHub_username>.github.io/<repository_name>/<path_to_file>
+```
+
+For example, if my GitHub username was `user`, my repo name was `repo`, and my repository had these files:
+
+```
+somerandomfolder/a.html
+contact.html
+index.html
+styles.css
+```
+
+Then the links to these files would be, respectively:
+
+```
+https://user.github.io/repo/somerandomfolder/a.html
+https://user.github.io/repo/somerandomfolder/contact.html
+https://user.github.io/repo/somerandomfolder or user.github.io/repo/somerandomfolder/index.html
+https://user.github.io/repo/somerandomfolder/styles.css
+```
+
+## Deploying a React project
+
+As simple as [following this tutorial](https://create-react-app.dev/docs/deployment#github-pages), from create-react-app, a well-known tool for, well, creating a React app.
+
+If you want routing, follow [this](https://www.freecodecamp.org/news/deploy-a-react-app-to-github-pages/). This involves making changes to your React project, so don't worry about touching GitHub for this.
+
+Maybe you might want to [automate the process with GitHub actions?](https://github.com/marketplace/actions/deploy-react-to-github-pages)
+
+**NOTE:** Please don't use this for Next.js projects; Vercel's already got you.
+
+## Deploying an Angular Project
+
+- [Here's your React equivalent.](https://github.com/angular-schule/angular-cli-ghpages)
+
+- [Want to automate this?](https://github.com/marketplace/actions/angular-deploy-gh-pages-actions)
+
+## Will I ever run out of bandwidth?
+
+There's a 100GB monthly *soft* bandwidth limit. That's the amount of data that can be transferred from GitHub to users accessing the site until GitHub starts getting a bit upset.
```

Comment:
It may be worth using more specific words than "getting a bit upset": What happens when we exceed the bandwidth limit? For example, does our site go down?